### DISCUSSION TITLE
Do the Repositories "helloworld" and "externals-howto" Serve the Same Purpose?

### DISCUSSION THREAD

**Discussion by charlesneimog** [pure-data/pure-data#2359] [Ideas]
Hi, minor thing, but don't the repositories https://github.com/pure-data/externals-howto and https://github.com/pure-data/helloworld serve the same purpose?

Perhaps we could merge one into the other?

**Comment by Spacechild1** [comment]
https://github.com/pure-data/helloworld is basically a template repo for writing your own externals, so merging it into https://github.com/pure-data/externals-howto would defeat the purpose IMO. However, the latter could include a link to the former.

**Reply by umlaeute** [reply]
> [https://github.com/pure-data/helloworld](https://github.com/pure-data/helloworld) is basically a template repo for writing your own externals

it's actually a template for writing externals with the `pd-lib-builder` buildsystem.

on the other hand, https://github.com/pure-data/externals-howto is documentation about writing externals.
the build-system is explicitly outside of the scope of the Externals HOWTO.
For practical reasons it does include a minimal build-system, which is `pd-lib-builder` (for decades it used the makefiles that come with Pd, but those are rather impractical to build on anything except SGI Iris)


> However, the latter could include a link to the former.

PRs are always welcome (though I would still like the Externals HOWTO to tread lightly on anything build-system related)
